nifi-nar-bundles/nifi-ignite-bundle/nifi-ignite-processors/src/main/java/org/apache/nifi/processors/ignite/cache/PutIgniteCache.java

```diff
@@ -0,0 +1,373 @@
+            igniteDataStreamer.perNodeBufferSize(perNodeBufferSize);
+            igniteDataStreamer.perNodeParallelOperations(perNodeParallelOperations);
+            igniteDataStreamer.autoFlushFrequency(autoFlushFrequency);
+            igniteDataStreamer.allowOverwrite(allowOverride);
+
+        } catch (Exception e) {
+            getLogger().error("Failed to schedule PutIgnite due to {}", new Object[] { e }, e);
+            throw new ProcessException(e);
+        }
+    }
+
+    /**
+     * Handle flow files
+     */
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        final int batchSize = context.getProperty(BATCH_SIZE).asInteger();
+        final List<FlowFile> flowFiles = session.get(batchSize);
+
+        if (flowFiles.isEmpty()) {
+            return;
+        }
+
+//        Map<String, byte[]> cacheMap = new java.util.HashMap<>();
+        List<Map.Entry<String, byte[]>> cacheItems = new ArrayList<>();
+        List<FlowFile> successfulFlowFiles = new ArrayList<>();
+        List<FlowFile> failedFlowFiles = new ArrayList<>();
+        try {
+            for (int i = 0; i < flowFiles.size(); i++) {
+                FlowFile flowFile = null;
+                try {
+                    flowFile = flowFiles.get(i);
+
+                    String key = flowFile.getAttribute(IGNITE_CACHE_ENTRY_KEY);
+
+                    if ( isFailedFlowFile(flowFile) ) {
+                        failedFlowFiles.add(flowFile);
+                        continue;
+                    }
+
+                    final byte[] byteArray = new byte[(int) flowFile.getSize()];
+                    session.read(flowFile, new InputStreamCallback() {
+                        @Override
+                        public void process(final InputStream in) throws IOException {
+                            StreamUtils.fillBuffer(in, byteArray, true);
+                        }
+                    });
+
+                    cacheItems.add(new AbstractMap.SimpleEntry<String,byte[]>(key, byteArray));
+                    successfulFlowFiles.add(flowFile);
+
+                } catch (Exception e) {
+                    getLogger().error("Failed to insert {} into IgniteDB due to {}", new Object[] { flowFile, e }, e);
+                    session.transfer(flowFile, REL_FAILURE);
+                }
+            }
+        } finally {
+            if (!cacheItems.isEmpty()) {
+                IgniteFuture<?> futures = igniteDataStreamer.addData(cacheItems);
+                Object result = futures.get();
+                getLogger().debug("Result {} of addData", new Object [] {result});
+            }
+
+            if (!successfulFlowFiles.isEmpty()) {
+                successfulFlowFiles = updateSuccessfulFlowFileAttributes(flowFiles, successfulFlowFiles, session);
+                session.transfer(successfulFlowFiles, REL_SUCCESS);
```

Comment:
The flow files are being processed in batches and I could not find a send method on provenance reporter that takes a batch of flow files. If there is a better way of handling this, please let me know.